packages/markdown/parseMarkdown.js

```diff
@@ -348,7 +348,7 @@ function createRender(context) {
 
       return [
         `<h${level} id="${hash}">`,
-        headingHtml,
+        `<a href="#${hash}" class="h2-heading">${headingHtml}</a>`,
         `<a aria-labelledby="${hash}" class="anchor-link" href="#${hash}" tabindex="-1">`,
         '<svg><use xlink:href="#anchor-link-icon" /></svg>',
         '</a>',
```

Comment:
What about using a single `<a />`?

Here is the DOM structure from Next Docs. For me, it makes more sense since it removes the need for the `aria-labelledby` since the textual description is already in the children. Plus we could bring back titles in the tabindex

![image](https://github.com/mui/material-ui/assets/45398769/f1e17781-942c-464a-9c97-a9862ae05ff3)

Reply:
this failed check in my pr is also because of this <a> tag ? (i am new to open source contribution)

Reply:
> What about using a single `<a />`?
> 
> Here is the DOM structure from Next Docs. For me, it makes more sense since it removes the need for the `aria-labelledby` since the textual description is already in the children. Plus we could bring back titles in the tabindex
> 
> ![image](https://user-images.githubusercontent.com/45398769/278040458-f1e17781-942c-464a-9c97-a9862ae05ff3.png)

@alexfauquette now can you see my changes (now I am using single `<a/>`)